tests/scripts/generate_tests_pk_can_do_psa.py

```diff
@@ -0,0 +1,366 @@
+            ],
+            'PSA_ALG_RSA_PSS(PSA_ALG_SHA_256)': [
+                'PSA_KEY_USAGE_VERIFY_HASH',
+            ],
+            'PSA_ALG_RSA_PKCS1V15_CRYPT': [
+                'PSA_KEY_USAGE_ENCRYPT',
+            ],
+            'PSA_ALG_RSA_OAEP(PSA_ALG_ANY_HASH)': [
+                'PSA_KEY_USAGE_ENCRYPT',
+            ],
+            'PSA_ALG_RSA_OAEP(PSA_ALG_SHA_256)': [
+                'PSA_KEY_USAGE_ENCRYPT',
+            ],
+        },
+    }
+}
+
+# PSA_WANT dependencies for key types and alg
+DEPENDS_ON = {
+    'MBEDTLS_PK_ECDSA': ['PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY'],
+    'MBEDTLS_PK_ECKEY': ['PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY'],
+    'MBEDTLS_PK_ECKEY_DH': ['PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY'],
+    'MBEDTLS_PK_ALG_ECDSA': ['PSA_HAVE_ALG_SOME_ECDSA'],
+    'PSA_ALG_SHA_256': ['PSA_WANT_ALG_SHA_256'],
+    'PSA_ALG_ECDH': ['PSA_WANT_ALG_ECDH'],
+    'MBEDTLS_PK_RSA': ['PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY'],
+    'PSA_ALG_RSA_PKCS1V15_SIGN': ['PSA_WANT_ALG_RSA_PKCS1V15_SIGN'],
+    'PSA_ALG_RSA_PSS': ['PSA_WANT_ALG_RSA_PSS'],
+    'PSA_ALG_RSA_PKCS1V15_CRYPT': ['PSA_WANT_ALG_RSA_PKCS1V15_CRYPT'],
+    'PSA_ALG_RSA_OAEP': ['PSA_WANT_ALG_RSA_OAEP'],
+}
+
+TEST_ALG = [
+    # Signature
+    'MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH)',
+    'MBEDTLS_PK_ALG_ECDSA(PSA_ALG_SHA_256)',
+    'PSA_ALG_RSA_PKCS1V15_SIGN(PSA_ALG_ANY_HASH)',
+    'PSA_ALG_RSA_PKCS1V15_SIGN(PSA_ALG_SHA_256)',
+    'PSA_ALG_RSA_PSS(PSA_ALG_ANY_HASH)',
+    'PSA_ALG_RSA_PSS(PSA_ALG_SHA_256)',
+    # Key agreement
+    'PSA_ALG_ECDH',
+    # Encryption
+    'PSA_ALG_RSA_PKCS1V15_CRYPT',
+    'PSA_ALG_RSA_OAEP(PSA_ALG_ANY_HASH)',
+    'PSA_ALG_RSA_OAEP(PSA_ALG_SHA_256)',
+]
+
+TEST_USAGE = [
+    'PSA_KEY_USAGE_ENCRYPT',
+    'PSA_KEY_USAGE_DECRYPT',
+    'PSA_KEY_USAGE_SIGN_HASH',
+    'PSA_KEY_USAGE_VERIFY_HASH',
+    'PSA_KEY_USAGE_DERIVE',
+    'PSA_KEY_USAGE_DERIVE_PUBLIC',
+]
+
+def populate_depends_on(*arg_list):
+    """
+    Generate the depends_on list based on the parameters that are passed. Check
+    DEPENDS_ON dictionary for the full list.
+    """
+    depends_on = []
+    known_deps = [dep for arg in arg_list
+                      for dep in DEPENDS_ON
+                      if dep in arg]
+    for dep in known_deps:
+        depends_on.extend(DEPENDS_ON[dep])
+    return depends_on
+
+def alg_include_hash(alg):
+    """
+    Return 1 if the algorithm that's passed in includes some hashing; 0 otherwise
+    """
+    return 1 if ('PSA_ALG_ANY_HASH' in alg) or ('PSA_ALG_SHA_' in alg) else 0
+
+def split_base_alg_from_hash(alg):
+    """
+    Separate the base algorithm and the hashing one and return them as touple:
+    - 1st item is the base alg;
+    - 2nd item is the hash alg or '' if the hash alg is not found.
+    """
+    base_alg = re.sub('\(.*\)', '', alg)
+    match = re.search(r'\((.*)\)', alg)
+    hash_alg = match.group(1) if match is not None else ''
+    return (base_alg, hash_alg)
+
+def are_algorithms_compatible(test_alg, key_alg):
```

Comment:
Sorry if I didn't mention this sooner, but for the PSA aspects (type/algorithm/usage compatibility, dependencies, pretty-printing, possibly even enumeration), there's existing code which is used to generate `test_suite_psa_crypto_storage_format.current.data` and `test_suite_psa_crypto.op_fail.generated.data`. The new thing here would be to throw the PK type into the mix for non-opaque test cases.

If this script stays and its output becomes an automatically generated file, then I think we should build on the existing PSA-oriented code. However, there's a good chance that before that, we'll work on getting rid of `mbedtls_pk_type_t`, and if we do that then there'll be a lot less work to do for `mbedtls_pk_can_do_psa` tests on top of what already exists for the PSA API tests.